### PR TITLE
Added isValid method and test of its class

### PR DESCRIPTION
**Pull request by gadirovakhadija**
Write isValid method an test class of it. In the test class I have some problems for just isValid's test method. please look isValid class first if its ok I'll do correct test class for it.

### CODE REVIEW

File: src/main/java/com/webperside/courseerpbackend/exception/OTPValidationException.java
Comment (hamidsultanzadeh) [inline src/main/java/com/webperside/courseerpbackend/exception/OTPValidationException.java:1]: we don't need that. we already have BaseException. You just need to add new enum on ErrorEnum

File: src/main/java/com/webperside/courseerpbackend/utils/OTPUtils.java [hunk 1]
```diff
@@ -1,12 +1,33 @@
 package com.webperside.courseerpbackend.utils;
 
+import com.webperside.courseerpbackend.exception.OTPValidationException;
+import com.webperside.courseerpbackend.models.enums.otp.OTPValidationResult;
+import com.webperside.courseerpbackend.services.redis.RedisService;
+import org.springframework.stereotype.Component;
+
 import java.util.Random;
 
 public class OTPUtils {
```
Comment (hamidsultanzadeh) [inline src/main/java/com/webperside/courseerpbackend/utils/OTPUtils.java:10]: add here @Component for triggering spring DI

File: src/main/java/com/webperside/courseerpbackend/utils/OTPUtils.java [hunk 2]
```diff
@@ -1,12 +1,33 @@
...
 
+    private static RedisService redisService;
+
+    public static void setRedisService(RedisService service){
```
Comment (hamidsultanzadeh) [inline src/main/java/com/webperside/courseerpbackend/utils/OTPUtils.java:14]: and also add a constructor for injecting RedisService. and take the injected value and set it to the static redis service. we did similar code block on PrivatePublicKeyUtils

Comment (hamidsultanzadeh) [reply src/main/java/com/webperside/courseerpbackend/utils/OTPUtils.java:14]: so you don't need this set method

File: src/main/java/com/webperside/courseerpbackend/utils/OTPUtils.java [hunk 3]
```diff
@@ -1,12 +1,33 @@
...
         return String.format("%04d", random.nextInt(10000));
     }
 
-}
+    public static void isValid(String key, String otp) {
+
+        String storedOtp = redisService.get(key);
+
+        if (storedOtp == null || !storedOtp.equals(otp)) {
+            throw new OTPValidationException(OTPValidationResult.OTP_IS_NOT_VALID);
```
Comment (hamidsultanzadeh) [inline src/main/java/com/webperside/courseerpbackend/utils/OTPUtils.java:28]: throw our exception

File: pom.xml [hunk 4]
```diff
@@ -22,6 +22,8 @@
 		<lombok-mapstruct-binding.version>0.2.0</lombok-mapstruct-binding.version>
 		<io.jsonwebtoken.jjwt.version>0.11.5</io.jsonwebtoken.jjwt.version>
 		<org.redisson.redisson-spring-boot-starter>3.23.4</org.redisson.redisson-spring-boot-starter>
+		<junit.version>4.13.1</junit.version>
+		<mockito-core.version>3.11.2</mockito-core.version>
```
Comment (hamidsultanzadeh) [inline pom.xml:26]: if not mistaken, we don't need these dependencies as well. because they have to be come with starter dependency of test

**Review by hamidsultanzadeh — CHANGES_REQUESTED** [review]

**Review by hamidsultanzadeh — APPROVED** [review]

**Review by aytiqaqash — APPROVED** [review]
LGTM

**Pull request merged**